File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.kt
class Solution {
    fun solution(id_list: Array<String>, report: Array<String>, k: Int): IntArray {
        val answer = IntArray(id_list.size) { 0 }
        val reportList = mutableMapOf<String, MutableSet<String>>()

        for (r in report) {
            val reporter = r.split(" ").first()
            val reportee = r.split(" ").last()

            if(reportList[reportee] == null) {
                reportList[reportee] = mutableSetOf(reporter)
            }else {
                reportList[reportee]!!.add(reporter)
            }
        }

        val listSet = reportList.filter { it.value.size >= k }.map { it.value }

        listSet.forEach {
            for (i in it) {
                answer[id_list.indexOf(i)]++
            }
        }
        return answer
    }
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.swift
func solution(_ id_list:[String], _ report:[String], _ k:Int) -> [Int] {
    var answer = [Int]()
    var reportees = [String: Int]()
    var reporters = [String: [String]]()
    
    Set(report).forEach {
        let info = $0.components(separatedBy: " ")
        let reporter = info[0], reportee = info[1]
        
        if let _ = reportees[reportee] {
            reportees[reportee]! += 1
        } else {
            reportees[reportee] = 1
        }
        
        if let _ = reporters[reporter] {
            reporters[reporter]!.append(reportee)
        } else {
            reporters[reporter] = [reportee]
        }
    }
    
    let filteredReportees = reportees.filter { $0.value >= k }.keys
    
    for id in id_list {
        var count = 0
        
        if let emailList = reporters[id] {
            emailList.forEach {
                if filteredReportees.contains($0) {
                    count += 1
                }
            }
        }
        
        answer.append(count)
    }
    
    return answer
}